CMakeLists.txt

```diff
@@ -22,24 +23,60 @@ endif()
 
 include (TestBigEndian)
 TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+
+configure_file(${PROJECT_SOURCE_DIR}/include/maxminddb_config.h.cmake.in 
+               ${PROJECT_SOURCE_DIR}/include/maxminddb_config.h)
+
+add_library(maxminddb
+  src/maxminddb.c
+  src/data-pool.c
+)
+add_library(maxminddb::maxminddb ALIAS maxminddb)
+
+target_compile_definitions(maxminddb PUBLIC PACKAGE_VERSION="${PROJECT_VERSION}")
+
 if(NOT IS_BIG_ENDIAN)
-  add_definitions(-DMMDB_LITTLE_ENDIAN=1)
+  target_compile_definitions(maxminddb PRIVATE MMDB_LITTLE_ENDIAN=1)
 endif()
 
 if(MSVC)
-    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+  target_compile_definitions(maxminddb PRIVATE _CRT_SECURE_NO_WARNINGS)
 endif()
 
 set(CMAKE_SHARED_LIBRARY_PREFIX lib)
 set(CMAKE_STATIC_LIBRARY_PREFIX lib)
 
-configure_file(include/maxminddb_config.h.cmake.in include/maxminddb_config.h)
+set(MAXMINDB_INCLUDE_DIR
+  .
+  include
+)
 
-include_directories(. src include)
+target_include_directories(maxminddb PUBLIC
+  $<BUILD_INTERFACE:${MAXMINDB_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>
+)
+
+set(HEADERS
+  include/maxminddb.h
+  include/maxminddb_config.h
+)
+set_target_properties(maxminddb PROPERTIES PUBLIC_HEADER "${MAXMINDB_HEADERS}")
 
-add_subdirectory(src)
-add_subdirectory(bin)
+install(TARGETS maxminddb
```

Comment:
In my testing, a `make install` no longer installs the header file or `mmdblookup`.

Before the change:

```
Install the project...
-- Install configuration: ""
-- Installing: /tmp/test/lib/libmaxminddb.a
-- Installing: /tmp/test/include/maxminddb_config.h
-- Installing: /tmp/test/include/maxminddb.h
-- Installing: /tmp/test/bin/mmdblookup
```

After:

```
Install the project...
-- Install configuration: ""
-- Installing: /tmp/test2/lib/libmaxminddb.a
-- Installing: /tmp/test2/lib/cmake/maxminddb/maxminddb-config.cmake
-- Installing: /tmp/test2/lib/cmake/maxminddb/maxminddb-config-noconfig.cmake
```

Reply:
I have fixed the install of the headers / let me check why the test program does not install anymore.

Reply:
The install works for me. I just had to configure cmake with -DMAXMINDDB_BUILD_TEST_PROGRAMS=ON / which is the default now.